sql/org_mozilla_firefox_derived/baseline_daily_v1/query.sql

```diff
@@ -0,0 +1,152 @@
+  FROM
+    org_mozilla_fenix_nightly.baseline AS baseline
+),
+--
+windowed AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    --
+    -- Take the earliest first_run_date if ambiguous.
+    MIN(first_run_date) OVER w1 AS first_run_date,
+    --
+    -- Sums over distinct baseline pings.
+    SUM(IF(duration BETWEEN 0 AND 100000, duration, 0)) OVER w1 AS durations,
+    --
+    -- Maintain lists of unique client activity dates for usage criteria based on client timestamps.
+    ARRAY_AGG(DATE(TIMESTAMP_SUB(end_time, INTERVAL duration SECOND))) OVER w1 AS start_dates,
+    ARRAY_AGG(DATE(end_time)) OVER w1 AS end_dates,
+    --
+    -- For all other dimensions, we use the mode of observed values in the day.
+    udf.mode_last(ARRAY_AGG(normalized_channel) OVER w1) AS normalized_channel,
+    udf.mode_last(ARRAY_AGG(android_sdk_version) OVER w1) AS android_sdk_version,
+    udf.mode_last(ARRAY_AGG(app_build) OVER w1) AS app_build,
+    udf.mode_last(ARRAY_AGG(app_channel) OVER w1) AS app_channel,
+    udf.mode_last(ARRAY_AGG(app_display_version) OVER w1) AS app_display_version,
+    udf.mode_last(ARRAY_AGG(architecture) OVER w1) AS architecture,
+    udf.mode_last(ARRAY_AGG(device_manufacturer) OVER w1) AS device_manufacturer,
+    udf.mode_last(ARRAY_AGG(device_model) OVER w1) AS device_model,
+    udf.mode_last(ARRAY_AGG(telemetry_sdk_build) OVER w1) AS telemetry_sdk_build,
+    udf.mode_last(ARRAY_AGG(locale) OVER w1) AS locale,
+    udf.mode_last(ARRAY_AGG(city) OVER w1) AS city,
+    udf.mode_last(ARRAY_AGG(country) OVER w1) AS country,
+    udf.mode_last(ARRAY_AGG(glean_baseline_locale) OVER w1) AS glean_baseline_locale,
+    udf.mode_last(ARRAY_AGG(normalized_os) OVER w1) AS os,
+    udf.mode_last(ARRAY_AGG(normalized_os_version) OVER w1) AS os_version,
+  FROM
+    base
+  WHERE
+    -- Reprocess all dates by running this query with --parameter=submission_date:DATE:NULL
+    (@submission_date IS NULL OR @submission_date = submission_date)
+  WINDOW
+    w1 AS (
+      PARTITION BY
+        sample_id,
+        client_id,
+        submission_date
+      ORDER BY
+        submission_timestamp
+      ROWS BETWEEN
+        UNBOUNDED PRECEDING
+        AND UNBOUNDED FOLLOWING
+    ),
+    -- We must provide a modified window for ROW_NUMBER which cannot accept a frame clause.
+    w1_unframed AS (
+      PARTITION BY
+        sample_id,
+        client_id,
+        submission_date
+      ORDER BY
+        submission_timestamp
+    )
+)
+--
+SELECT
+  wnd.submission_date,
+  cfs.fenix_first_seen_date,
+  cfs.fennec_first_seen_date,
+  wnd.* EXCEPT (_n, submission_date)
+  -- We need to do some post-processing on these arrays; they may contain nulls,
+  -- but BigQuery will throw an error if it tries to materialize an array with
+  -- null elements.
+  REPLACE(
+    ARRAY(SELECT DISTINCT d FROM UNNEST(start_dates) d WHERE d IS NOT NULL) AS start_dates,
```

Comment:
@relud Do you have an opinion on whether it would be better to save this is a bit pattern in baseline_daily rather than an array of dates? We'd be able to use a `BIT_OR` rather than an ARRAY_AGG, and it saves us this extra step of having to filter out null values.

Reply:
those seem like good reasons to use a bit pattern.

Is it known who or what will be using this field, and how?

Would the pattern would be relative to submission_date?

Would we be exposing the bit pattern as a date array in a view?

I'm fine with it, but I'm also a bad judge of how confusing bit patterns are for consumers.